doc/sphinx/elaboration.rst

```diff
@@ -0,0 +1,30 @@
+.. _elaboration:
+
+===========
+Elaboration
+===========
+
+The elaboration process is what allows users to write in a richer and
```

Comment:
Maybe:
Elaboration extends the language accepted by the Coq kernel to make it easier to use.  For example, this lets the user omit most type annotations because they can be inferred, ...

What user-visible parts of the system use only the core language?  `Show Proof` shows core or extended?

Reply:
No user-visible part of the system only use the core language, expect when using the `Set Printing All` option (or equivalent "Display all low-level contents" menu in CoqIDE). For instance,

```coq
Goal 1 = 1.
Proof.
reflexivity.
Show Proof.
```

prints just `eq_refl`. When activating the low-level printing, it becomes `(@eq_refl nat (S O))`.

Reply:
As @mattam82 commented in his notes of our meeting at the last CUDW, we should ideally be able to disable all high-level features on parsing and not just on printing, to be able to show examples where we are guaranteed to only rely on the core language.

Reply:
How about my wording suggestion?

> Maybe:
Elaboration extends the language accepted by the Coq kernel to make it easier to use. For example, this lets the user omit most type annotations because they can be inferred, ...

Reply:
Thanks for repeating it. I overlooked it because I thought I had answered to this comment already.

Reply:
Cf. #11646